src/main/java/calculator/Operator.java

```diff
@@ -0,0 +1,53 @@
+package calculator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Operator {
+
+  private final String operator;
+  private static final Map<String, Operator> operatorMap = new HashMap<>();
+
+  private Operator(String operator) {
+    this.operator = operator;
+    operatorMap.put(this.operator, this);
+  }
+
+  abstract int calculate(int operand1, int operand2);
+
+  public static Operator PLUS = new Operator("+") {
```

Comment:
상수 인스턴스인것 같으니 final 키워드를 사용해서 상수임을 나타내는 것이 좋겠습니다. 😁 
```suggestion
  public final static Operator PLUS = new Operator("+") {
```